platform/api/aws/images.go

```diff
@@ -375,8 +375,15 @@ func (a *API) createImage(params *ec2.RegisterImageInput) (string, error) {
 	res, err := a.ec2.RegisterImage(params)
 
 	if err == nil {
+		err = a.CreateTags([]string{*res.ImageId}, map[string]string{
```

Comment:
I think it does for snapshots. The code that tags a snapshot runs unconditionally, even if the snapshot is "done" / "complete" https://github.com/coreos/mantle/blob/d49d04a92ac30ffb74b77b6bf139cb782c02a530/platform/api/aws/images.go#L258-L264

Can you reference where it doesn't pick up tagging given a poorly timed SIGTERM?

Reply:
Right, I was initially looking at how `FindSnapshot` returns early (and thus skips the post-processing) if we don't fall back to searching for pre-existing import tasks:

https://github.com/coreos/mantle/blob/d49d04a92ac30ffb74b77b6bf139cb782c02a530/platform/api/aws/images.go#L126-L132

But actually, we'd only get there if the `Name` tag was already correct judging by the filters. So I think you're right, we do implicitly do this for snapshots just from how we search for them in the first place. If it wasn't tagged, then the import task fallback will pick it up and complete it (assuming the task wasn't GC'ed already).

Reply:
Yup, reading it again it's certainly non-obvious that FindSnapshot works that way.

I think it's up to you whether we make this idempotent/resumable or not.
As long as there's only async cleanup tasks that rely on the name tag, not plume itself, I don't think it really matters that much.

Reply:
This code runs in the `plume prerelease` (i.e., CL release) path, even though CL releases don't actually need the tag.  Since the rest of the code in that path tries pretty hard to be idempotent, I'd be in favor of also doing so here if possible.

Reply:
I updated this to always tag pre-existing images! :arrow_down: